lib/active_record/connection_adapters/fibered_mysql2_adapter.rb

```diff
@@ -1,30 +1,19 @@
 # frozen_string_literal: true
 
-require 'em-synchrony'
 require 'active_model'
 require 'active_record/errors'
-require 'active_record/connection_adapters/em_mysql2_adapter'
+require 'active_record/connection_adapters/mysql2_adapter'
 
 module FiberedMysql2
-  module FiberedMysql2Adapter_4_2
+  module FiberedMysql2Adapter_6
     def lease
```

Comment:
(nit) It looks a lot like https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L298-L365 except the use of `Fiber.current` over `Thread.current`.

Any particular reason why we cannot just adapt as is from the rails source code, which certainly feels a bit more readable with the use of `in_use?`

Reply:
I started there years ago.

But these methods that look at `in_use?` followed by `@owner` kept tripping me up as I studied them, because `in_use?` is just an alias that returns `@owner`, and asking for the owner twice actually creates a risk of a race condition.